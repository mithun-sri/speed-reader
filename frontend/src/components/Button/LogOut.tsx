import { Box, IconButton } from "@mui/material";
import { useLogout } from "../../hooks/users";

const LogOut: React.FC<{
  fontSize: number;
}> = ({ fontSize }) => {
  const logoutMutation = useLogout();

  const handleLogoutClick = async () => {
    try {
      await logoutMutation.mutateAsync();
    } catch (error) {
      console.error("Error logging out:", error);
    }
  };

  return (
    <IconButton onClick={handleLogoutClick}>
      <Box
        sx={{
          fontFamily: "JetBrains Mono, monospace",
          color: "#fff",
          backgroundColor: "#323437",
          fontWeight: "bolder",
          fontSize: fontSize,
          padding: "4px 8px",
          borderRadius: "20px",
        }}
      >
        Log Out
      </Box>
    </IconButton>
  );
};

export default LogOut;
