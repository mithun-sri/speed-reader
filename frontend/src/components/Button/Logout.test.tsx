import { faker } from "@faker-js/faker";
import renderer, { ReactTestRendererJSON } from "react-test-renderer";
import LogOut from "./LogOut";

it("Matches snapshot", () => {
  const fontSize = faker.number.int({ min: 10, max: 100 });
  const component = renderer.create(<LogOut fontSize={fontSize} />);
  const tree = component.toJSON() as ReactTestRendererJSON;
  expect(tree).toMatchSnapshot();
});
