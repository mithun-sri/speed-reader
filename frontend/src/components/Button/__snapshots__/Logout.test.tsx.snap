// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<button
  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
  disabled={false}
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <div
    className="MuiBox-root css-14ffi6j"
  >
    Log Out
  </div>
</button>
`;
